File: Lib/Telerik/Examples/QuickStart/GridView/Filtering/CustomFilterRow/GridFilteringChildRowElement.vb
Imports Telerik.WinControls.UI
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.GridView.Filtering.CustomFilterRow
	Public Class GridFilteringChildRowElement
		Inherits GridRowElement
		#Region "Fields"

		Private cellElement As GridCellElement
		Private hostedControl As AdvanceFilterControl

		#End Region

		#Region "Initialization"

		Public Sub New()
		End Sub

		Public Overrides Sub Initialize(ByVal rowInfo As GridViewRowInfo)
			MyBase.Initialize(rowInfo)

			rowInfo.SuspendPropertyNotifications()
			rowInfo.Height = Me.hostedControl.Size.Height
			rowInfo.ResumePropertyNotifications()
		End Sub

		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			Me.cellElement = New GridCellElement(Nothing, Me)
			Me.cellElement.StretchHorizontally = True
			Me.cellElement.StretchVertically = True
			Me.Children.Add(cellElement)

			Me.hostedControl = New AdvanceFilterControl()
			Dim hostItem As New RadHostItem(Me.hostedControl)
			Me.cellElement.Children.Add(hostItem)
		End Sub

		Protected Overrides Sub OnLoaded()
			MyBase.OnLoaded()
			Dim grid As RadGridView = Me.GridControl
			Me.hostedControl.GridView = grid
		End Sub

		#End Region

		#Region "Methods"

		Public Overrides Function IsCompatible(ByVal data As GridViewRowInfo, ByVal context As Object) As Boolean
			Return TypeOf data Is GridViewFilteringChildRowInfo
		End Function

		#End Region
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/GridView/Filtering/CustomFilterRow/GridFilteringChildRowElement.cs
using Telerik.WinControls.UI;
using Telerik.WinControls;

namespace Telerik.Examples.WinControls.GridView.Filtering.CustomFilterRow
{
    public class GridFilteringChildRowElement : GridRowElement
    {
        #region Fields

        private GridCellElement cellElement;
        private AdvanceFilterControl hostedControl;

        #endregion

        #region Initialization

        public GridFilteringChildRowElement()
        {
        }

        public override void Initialize(GridViewRowInfo rowInfo)
        {
            base.Initialize(rowInfo);

            rowInfo.SuspendPropertyNotifications();
            rowInfo.Height = this.hostedControl.Size.Height;
            rowInfo.ResumePropertyNotifications();
        }

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            this.cellElement = new GridCellElement(null, this);
            this.cellElement.StretchHorizontally = true;
            this.cellElement.StretchVertically = true;
            this.Children.Add(cellElement);

            this.hostedControl = new AdvanceFilterControl();
            RadHostItem hostItem = new RadHostItem(this.hostedControl);
            this.cellElement.Children.Add(hostItem);
        }

        protected override void OnLoaded()
        {
            base.OnLoaded();
            RadGridView grid = this.GridControl;
            this.hostedControl.GridView = grid;
        }

        #endregion

        #region Methods

        public override bool IsCompatible(GridViewRowInfo data, object context)
        {
            return data is GridViewFilteringChildRowInfo;
        }

        #endregion
    }
}
